Mage/src/main/java/mage/abilities/effects/common/continuous/SetPowerToughnessSourceEffect.java

```diff
@@ -13,41 +14,66 @@
 
 /**
  *
- * @author BetaSteward_at_googlemail.com, North
+ * @author BetaSteward_at_googlemail.com, North, Alex-Vasile
  */
 public class SetPowerToughnessSourceEffect extends ContinuousEffectImpl {
 
-    private DynamicValue amount;
-    private int power;
-    private int toughness;
+    private DynamicValue power;
+    private DynamicValue toughness;
+    // If true, the base power/toughness are net, otherwise the final value is set.
+    private boolean changeBaseValue;
```

Comment:
we should probably have a `switchPowerToughness` method so we can encapsulate all that

Reply:
What class would you encapsulate them into? It can't be `MageInt` because a permanent's power and its toughness are two completely separate `MageInt` objects.

Reply:
probably `Permanent` since it's only ever for a permanent

Reply:
I will see about the setBoostedValue usage first, I will look into encapsulating after. But it would make sense to put in `Permanent`.

Reply:
Alright, give that a look over